jobs/kpi-forecasting/kpi-forecasting/models/prophet_forecast.py

```diff
@@ -0,0 +1,119 @@
+import json
+import pandas as pd
+import prophet
+
+from datetime import datetime
+from dataclasses import dataclass
+from models.base_forecast import BaseForecast
+from typing import Dict
+
+
+@dataclass
+class ProphetForecast(BaseForecast):
+    @property
+    def column_names_map(self) -> Dict[str, str]:
+        return {"submission_date": "ds", "value": "y"}
+
+    def _fit(self) -> None:
+        """
+        Fit a Prophet model using the `observed_df` that was generated using
+        Metric Hub. This method updates `self.model`.
+        """
+        self.model = prophet.Prophet(
+            **self.parameters,
+            uncertainty_samples=self.number_of_simulations,
+            mcmc_samples=0,
+        )
+
+        if self.use_holidays:
+            self.model.add_country_holidays(country_name="US")
+
+        # Modify observed data to have column names that Prophet expects, and fit
+        # the model
+        self.model.fit(self.observed_df.rename(columns=self.column_names_map))
+
+    def _predict(self) -> pd.DataFrame:
+        """
+        Forecast using `self.model`.
+        """
+        # generate the forecast samples
+        samples = self.model.predictive_samples(
+            self.dates_to_predict.rename(columns=self.column_names_map)
+        )
+        df = pd.DataFrame(samples["yhat"])
+
+        # add dates_to_predict to the output
+        df["submission_date"] = self.dates_to_predict
+
+        return df
+
+    def _predict_legacy(self) -> pd.DataFrame:
+        """
+        Recreate the legacy format used in
+        `moz-fx-data-shared-prod.telemetry_derived.kpi_automated_forecast_v1`.
+        """
+        # TODO: This method should be removed once the forecasting data model is updated:
+        # https://docs.google.com/document/d/18esfJraogzUf1gbZv25vgXkHigCLefazyvzly9s-1k0.
+
+        df = self.model.predict(
+            self.dates_to_predict.rename(columns=self.column_names_map)
+        )
+
+        # set legacy column values
```

Comment:
Comments should provide understanding not just state what the code does. In this case, it would be more valuable to understand why this is needed. What do we mean by legacy?

Reply:
I made an effort to explain this in the PR description, but maybe that's not the best place for the explanation. I also added the following description to the readme:

> There is [an ongoing project](https://docs.google.com/document/d/18esfJraogzUf1gbZv25vgXkHigCLefazyvzly9s-1k0/edit) to re-model the way we store forecast data in Big Query. One key change in this refactor is to format output data to be as close as possible to what we expect the new data model to look like, while also preserving the legacy data model for the time being. Methods related to legacy outputs are clearly marked in the code (usually including legacy in the method name) and are commented with TODO items that indicate which code paths can be removed once the transition has been made.

Reply:
If this is specific to the code, I'd include that information with the code. It's much easier to read and understand, I may look at the code without knowing that README has code specific comments.

Reply:
I removed this from the README in favor of including a link to the Jira ticket in the comments.